skfp/utils/splitters.py

```diff
@@ -0,0 +1,630 @@
+    scaffolds from the training set.
+
+    The split is fully deterministic, with the smallest scaffold sets assigned to the test or validation
+    subset and the rest to the training subset.
+
+    The split fractions (train_size, test_size) must sum to 1.
+
+    Parameters
+    ----------
+    data : sequence
+        Sequence representing either SMILES strings or RDKit `Mol` objects.
+
+    additional_data: sequence
+        Additional sequences to be split alongside the main data (e.g., labels or feature vectors).
+
+    train_size : float, default=None
+        The fraction of data to be used for the train subset. If None, it is set to 1 - test_size - valid_size.
+        If valid_size is not provided, train_size is set to 1 - test_size. If train_size, test_size and
+        valid_size aren't set, train_size is set to 0.8.
+
+    test_size : float, default=None
+        The fraction of data to be used for the validation subset. If None, it is set to 1 - train_size - valid_size.
+        If valid_size is not provided, test_size is set to 1 - train_size. If train_size, test_size and
+        valid_size aren't set, test_size is set to 0.1.
+
+    valid_size : float, default=None
+        The fraction of data to be used for the test subset. If None, it is set to 1 - train_size - valid_size.
+        If train_size, test_size and valid_size aren't set, train_size is set to 0.1.
+
+    include_chirality: bool, default=False
+        Whether to take chirality of molecules into consideration.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit `Mol` objects, or only the indices of the subsets.
+
+    Returns
+    ----------
+    subsets : tuple[list, list, list]
+        A tuple of train, validation, and test subsets. The format depends on `return_indices`:
+        - if `return_indices` is False, returns lists of SMILES strings
+        or RDKit `Mol` objects depending on the input
+        - if `return_indices` is True, returns lists of indices
+    additional_data_splitted: list[sequence]:
+        the method may return any additional data splitted in the same way as
+        provided SMILES
+
+    References
+    ----------
+    .. [1] `Bemis, G. W., & Murcko, M. A.
+        "The properties of known drugs. 1. Molecular frameworks."
+        Journal of Medicinal Chemistry, 39(15), 2887-2893.
+        https://www.researchgate.net/publication/14493474_The_Properties_of_Known_Drugs_1_Molecular_Frameworks`_
+
+    .. [2] `Z. Wu, B. Ramsundar, E. N. Feinberg, J. Gomes, C. Geniesse, A. S. Pappu, K. Leswing, V. Pande
+        "MoleculeNet: A Benchmark for Molecular Machine Learning."
+        Chemical Science, 9(2), 513-530.
+        https://www.researchgate.net/publication/314182452_MoleculeNet_A_Benchmark_for_Molecular_Machine_Learning`_
+
+    """
+    _validate_split_sizes(train_size, test_size)
```

Comment:
This has slightly other logic than for train-valid-test. Probably this function can be removed and put directly here and in the train-test function